doc/interfaces/serial.rst

```diff
@@ -13,5 +22,80 @@ Bus
 
 Internals
 ---------
+The frame that will be send and receive over the serial interface consists of
+six parts. The start and the stop byte for the frame, the timestamp, DLC,
+arbitration ID and the payload. Only the payload has a variable length of bytes
+(0 - 8 bytes), the other parts are fix. Both, the timestamp and the arbitration
+ID will be interpreted as 4 byte unsigned integers. The DLC is also a unsigned
+integer with a length of 1 byte.
+
+Serial frame format
+^^^^^^^^^^^^^^^^^^^
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+|                   | Start of frame | Timestamp                                     | DLC                           | Arbitration ID          | Payload | End of frame |
++===================+================+==============================+================+===============================+=========================+=========+==============+
+| **Length (Byte)** | 1              | 4                                             | 1                             | 4                       | 0 - 8   | 1            |
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+| **Data type**     | Byte           | Unsigned 4 byte integer                       | Unsigned 1 byte integer       | Unsigned 4 byte integer | Byte    | Byte         |
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+| **Byte order**    | \-             | Little-Endian                                 | Little-Endian                 | Little-Endian           | \-      | \-           |
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+| **Description**   | Must be 0xAA   | Usually s, ms or µs since start of the device | Length in byte of the payload | \-                      | \-      | Must be 0xBB |
```

Comment:
Is this implemented for some specific device or is it just something custom that is supposed to be implemented on the other side? In that case I think it should be clear how the protocol is specified. Right now it is in seconds but it should probably be in milli or microseconds which means it must be 8 bytes.

Also we might just as well add support for extended/std and remote frames to avoid backwards incompatible changes in the future. Add one byte with various flags.